src/main/modules/pcq/controller.ts

```diff
@@ -0,0 +1,54 @@
+import { AxiosError } from 'axios';
+import config from 'config';
+import { Response } from 'express';
+
+import { FormFieldsFn } from '../../../main/app/form/Form';
+import { AppRequest } from '../../app/controller/AppRequest';
+import PayAndSubmitPostController from '../../steps/c100-rebuild/check-your-answers/PayAndSubmitPostController';
+import ResponseSummaryConfirmationPostController from '../../steps/tasklistresponse/summary/postController';
+
+import { PCQProvider } from './index';
+
+export class PcqController {
+  protected static handleError(error: string | AxiosError, res: Response, redirectUrl: string): void {
+    PCQProvider.log('error', error);
+    return res.redirect(redirectUrl);
+  }
+
+  async launch(req: AppRequest, res: Response, returnUrl: string): Promise<void> {
+    try {
+      PCQProvider.init(req);
+      const url = config.get('services.equalityAndDiversity.url');
+      const path: string = config.get('services.equalityAndDiversity.path');
+      const status = await PCQProvider.service.getPcqHealthStatus(`${url}/health`);
+
+      if (status === 'UP') {
+        try {
+          const pcqServiceUrl = await PCQProvider.getPcqServiceUrl(url as string, path, req, returnUrl);
```

Comment:
When can getPcqServiceUrl fail?

Reply:
During token generation

Reply:
But if that's the case, where is the error handled while token generation. Can you spot the place where it can fail?